packages/charts/src/chart_types/metric/renderer/dom/text_measurements.tsx

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { getTextParts, TextParts } from './text_processing';
+import { DEFAULT_FONT_FAMILY } from '../../../../common/default_theme_attributes';
+import { Font } from '../../../../common/text_utils';
+import { TextMeasure, withTextMeasure } from '../../../../utils/bbox/canvas_text_bbox_calculator';
+import { clamp, isNil, LayoutDirection } from '../../../../utils/common';
+import { Size } from '../../../../utils/dimensions';
+import { wrapText } from '../../../../utils/text/wrap';
+import { MetricStyle } from '../../../../utils/themes/theme';
+import { MetricDatum, isMetricWProgress, MetricWNumber } from '../../specs';
+
+interface HeightBasedSizes {
+  iconSize: number;
+  titleFontSize: number;
+  subtitleFontSize: number;
+  extraFontSize: number;
+  valueFontSize: number;
+  valuePartFontSize: number;
+}
+
+type BreakPoint = 'xxxs' | 'xxs' | 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl';
+
+type ElementVisibility = {
+  titleMaxLines: number;
+  subtitleMaxLines: number;
+  title: boolean;
+  subtitle: boolean;
+  extra: boolean;
+};
+/**
+ * synced with scss variables
+ */
+const PROGRESS_BAR_WIDTH = 10;
```

Comment:
```suggestion
// synced with scss variables
const PROGRESS_BAR_WIDTH = 10;
```

This was a general comment not a JSdocs for this value